modules/tpmnBidAdapter.js

```diff
@@ -1,164 +1,248 @@
+
+  if (bid.params.inventoryId) rtbData.ext = {};
+  if (bid.params.inventoryId) rtbData.ext.inventoryId = bid.params.inventoryId
+
+  const ortb2Data = bidderRequest?.ortb2 || {};
+  const bcat = ortb2Data?.bcat || bid.params.bcat || [];
+  const badv = ortb2Data?.badv || bid.params.badv || [];
+  const bapp = ortb2Data?.bapp || bid.params.bapp || [];
+
+  if (bcat.length > 0) {
+    rtbData.bcat = bcat;
+  }
+  if (badv.length > 0) {
+    rtbData.badv = badv;
+  }
+  if (badv.length > 0) {
+    rtbData.bapp = bapp;
+  }
+
+  return {
+    method: 'POST',
+    url: BIDDER_ENDPOINT_URL + '?v=' + ADAPTER_VERSION,
+    data: rtbData,
+    options: { contentType: 'application/json;charset=UTF-8', withCredentials: false }
+  }
+}
+
+function interpretResponse(response, request) {
+  return CONVERTER.fromORTB({ request: request.data, response: response.body }).bids;
+}
+
+registerBidder(spec);
+
+const CONVERTER = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: DEFAULT_BID_TTL,
+    currency: DEFAULT_CURRENCY
   },
-  /**
-   * Unpack the response from the server into a list of bids.
-   *
-   * @param {serverResponse} serverResponse A successful response from the server.
-   * @return {Bid[]} An array of bids which were nested inside the server.
-   */
-  interpretResponse: function (serverResponse, serverRequest) {
-    if (!Array.isArray(serverResponse.body)) {
-      return [];
+  imp(buildImp, bidRequest, context) {
+    let imp = buildImp(bidRequest, context);
+    if (!imp.bidfloor && bidRequest.params.bidFloor) {
+      imp.bidfloor = bidRequest.params.bidFloor;
     }
-    // server response body is an array of bid results
-    const bidResults = serverResponse.body;
-    // our server directly returns the format needed by prebid.js so no more
-    // transformation is needed here.
-    return bidResults;
+    [VIDEO, BANNER].forEach(namespace => {
+      COMMON_PARAMS.forEach(param => {
+        if (bidRequest.params.hasOwnProperty(param)) {
+          utils.deepSetValue(imp, `${namespace}.${param}`, bidRequest.params[param])
+        }
+      })
+    })
+    return imp;
   },
+  bidResponse(buildBidResponse, bid, context) {
+    const {bidRequest} = context;
+    const bidResponse = buildBidResponse(bid, context);
 
-  getUserSyncs: function (syncOptions, serverResponses, gdprConsent, uspConsent) {
-    const syncArr = [];
-    if (syncOptions.iframeEnabled) {
-      let policyParam = '';
-      if (gdprConsent && gdprConsent.consentString) {
-        if (typeof gdprConsent.gdprApplies === 'boolean') {
-          policyParam += `&gdpr=${Number(gdprConsent.gdprApplies)}&gdpr_consent=${gdprConsent.consentString}`;
-        } else {
-          policyParam += `&gdpr=0&gdpr_consent=${gdprConsent.consentString}`;
-        }
-      }
-      if (uspConsent && uspConsent.consentString) {
-        policyParam += `&ccpa_consent=${uspConsent.consentString}`;
+    utils.logWarn('Building bidResponse');
```

Comment:
I'm not sure if I missed these in my previous review - I don't think they belong here, perhaps this was to help during development? Prebid already logs requests and responses.

Reply:
I added it while creating example.html, and it was reflected as is. I will remove that part.